src/parse/lexer.rs

```diff
@@ -506,14 +506,18 @@ where
     }
 
     // Lex a hex/octal/decimal/binary number without a decimal point.
-    fn lex_number_radix(&mut self, start_pos: usize, radix: u32) -> LexResult {
+    fn lex_number_radix(&mut self, start_pos: usize, radix: u32, prefix: &str) -> LexResult {
         let num = self.radix_run(radix);
-        let value = match radix {
-            10 => num,
-            x => format!("{}#{}", x, num),
-        };
-        let end_pos = self.get_pos();
-        Ok((start_pos, Tok::Int { value }, end_pos))
+        if radix < 16 && Lexer::<T>::is_digit_of_radix(self.chr0, 16) {
+            Err(LexicalError {
+                error: LexicalErrorType::DigitOutOfRadix,
+                location: self.get_pos(),
+            })
+        } else {
+            let value = format!("{}{}", prefix, num);
+            let end_pos = self.get_pos();
+            Ok((start_pos, Tok::Int { value }, end_pos))
+        }
```

Comment:
Brain fart, I new I had to add back the prefix but added the erlang prefix rather than the gleam